### PR TITLE
Create crawl / alpha_crawl.py

### PR DESCRIPTION
**Pull request by ioanvl**

### CODE REVIEW

File: crawler/alpha_vantage/alpha_crawl.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+from ..common import *
+from contracts import *
+import time 
+import json
+from discord import *
+import pandas
```
Comment (Zarkopafilis) [inline crawler/alpha_vantage/alpha_crawl.py]: add this to requirements.txt

File: crawler/alpha_vantage/alpha_crawl.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+from ..common import *
```
Comment (Zarkopafilis) [inline crawler/alpha_vantage/alpha_crawl.py]: register and run this through the crawler microframework

File: crawler/alpha_vantage/alpha_crawl.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+from contracts import *
+import time 
+import json
+from discord import *
+import pandas
+
+def get_stock(name, api_key):
+    eq_url="https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&symbol={}&outputsize=full&apikey={}&datatype=csv".format(name, api_key)
```
Comment (Zarkopafilis) [inline crawler/alpha_vantage/alpha_crawl.py]: abstract as config

File: crawler/alpha_vantage/alpha_crawl.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                name,
+                row['timestamp'],
+                [str(row['open']), str(row['close']), str(row['high']), str(row['low'])]
+            )
+        )
+    
+    data_list = CrawledDataListRequest(post_list)
+
+    try:
+        response = post_social_data(data_list)
```
Comment (Zarkopafilis) [inline crawler/alpha_vantage/alpha_crawl.py]: this is the wrong endpoint. create one for econometrics as shown before and integrate

File: crawler/alpha_vantage/alpha_crawl.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import json
+from discord import *
+import pandas
+
+def get_stock(name, api_key):
+    eq_url="https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&symbol={}&outputsize=full&apikey={}&datatype=csv".format(name, api_key)
+
+    try:
+        temp_data=pandas.read_csv(eq_url)
+        warning_update("{} equity could not be downloaded".format(name))
```
Comment (deepettas) [inline crawler/alpha_vantage/alpha_crawl.py]: Why is this warning update inside the try: and not in the except part?

File: bash_conda-pip_install [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+while read requirement; do conda install --yes $requirement || pip install $requirement; done < requirements.txt
```
Comment (Zarkopafilis) [inline bash_conda-pip_install]: Remove this. Achievable with pip install -r requirements.txt

File: contracts/EconDataList.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
+from typing import List
+from .EconDataRecord import *
+import json
+
+class EconDataListRequest(object):
+    def __init__(self, data: List[EconData]) -> None:
```
Comment (Zarkopafilis) [inline contracts/EconDataList.py]: Rename this. @deepettas What about EconometricsDataRequest

File: controllers/crawler.py [hunk 8]
```diff
@@ -62,6 +63,27 @@ def insert_data():
...
+def insert_econ_data():
+    req = EconDataListRequest.from_json(json.loads(flask.request.data))
+
+    if req is None or req.data is None or len(req.data) == 0:
+        return json.dumps({ error: ''})
+    # validate
+    for record in req.data:
+        if record is None or record.timestamp is None or record.equity is None:
+            return error()
+        #=============================================
```
Comment (Zarkopafilis) [inline controllers/crawler.py]: Don't blindly copy and paste. Add proper validation and remove redundant comments

File: crawler/alpha_vantage/alpha_crawl.py [hunk 9]
```diff
@@ -10,8 +12,8 @@ def get_stock(name, api_key):
 
     try:
         temp_data=pandas.read_csv(eq_url)
-        warning_update("{} equity could not be downloaded".format(name))
     except:
+        warning_update("{} equity could not be downloaded".format(name))
```
Comment (Zarkopafilis) [inline crawler/alpha_vantage/alpha_crawl.py]: Make this an error

**Pull request merged**